tsl/src/compression/compression.c

```diff
@@ -1364,13 +1371,13 @@ tsl_compressed_data_decompress_reverse(PG_FUNCTION_ARGS)
 	if (PG_ARGISNULL(0))
 		PG_RETURN_NULL();
 
-	compressed = PG_GETARG_DATUM(0);
-	header = (CompressedDataHeader *) PG_DETOAST_DATUM(compressed);
 	if (SRF_IS_FIRSTCALL())
 	{
 		funcctx = SRF_FIRSTCALL_INIT();
 		oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
 
+		header = get_compressed_data_header(PG_GETARG_DATUM(0));
+
 		iter =
 			definitions[header->compression_algorithm]
 				.iterator_init_reverse(PG_GETARG_DATUM(0), get_fn_expr_argtype(fcinfo->flinfo, 1));
```

Comment:
I didn't understand why this is not `.iterator_init_reverse(PointerGetDatum(header), get_fn_expr_argtype(fcinfo->flinfo, 1));`

Reply:
you are right, your suggestion is better cause it prevents  detoasting twice